ethcore/src/client/client.rs

```diff
@@ -1511,6 +1511,49 @@ impl CallContract for Client {
 			.map_err(|e| format!("{:?}", e))
 			.map(|executed| executed.output)
 	}
+
+	fn call_contract_before(&self, header: &Header, address: Address, data: Bytes) -> Result<Bytes, String> {
+		let db = self.state_db.read().boxed_clone();
+
+		// early exit for pruned blocks
+		if db.is_prunable() && self.pruning_info().earliest_state > header.number() {
+			return Err(CallError::StatePruned.to_string());
+		}
+
+		let parent = self.chain.read().block_header_data(header.parent_hash()).ok_or_else(|| {
+			error!(target: "client", "Failed to call contract at {:?}'s child", header.parent_hash());
+			format!("Failed to call contract at {:?}'s child", header.parent_hash())
+		})?;
+		let root = parent.state_root();
+		let nonce = self.engine.account_start_nonce(header.number());
+		let mut state = State::from_existing(db, root, nonce, self.factories.clone())
+			.map_err(|_| CallError::StatePruned.to_string())?;
+
+		let from = Address::default();
+		let transaction = transaction::Transaction {
```

Comment:
I think this merits a comment: looks like it's a "fake" tx used to satisfy the type signature of `do_virtual_call`? Maybe it can be extracted to its own method to make the intent clear?

Reply:
@dvdplm Well, this is interesting. This function has nothing to do with the PR. It is from the stable  branch: https://github.com/paritytech/parity-ethereum/blob/bf74eb5cc891f6c8820aaee8ad11ac6029c26ab6/ethcore/src/client/client.rs#L1246.

Would you like to refactor it?